app/Services/Network/SafeHttp.php

```
@@ -3,78 +3,144 @@
     }
 
     /**
-     * Build a PSR-18 client that applies the same per-redirect-hop validation
-     * via Guzzle middleware. Use for libraries that take an injected client
-     * (e.g. PhanAn\Poddle\Poddle::fromUrl()).
+     * Build a PSR-18 client whose handler stack applies the same hop-by-hop
+     * pinning + validation as the Laravel Http methods above. Use for libraries
+     * that take an injected client (e.g. PhanAn\Poddle\Poddle::fromUrl()).
      */
-    public function guzzleClient(int $timeoutInSeconds = 30): Client
+    public function getPinnedGuzzleClient(string $url, bool $trackRedirects = false, int $timeoutInSeconds = 30): Client
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$url'. (undefined)

(UnusedFormalParameter)